### PR TITLE
Mat 5381 refactor human readable generation

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-5381](https://jira.cms.gov/browse/MAT-5381)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/controllers/EffectiveDataRequirementController.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  @PutMapping(
+      value = "/effective-data-requirements",
+      produces = {MediaType.APPLICATION_JSON_VALUE},
+      consumes = {MediaType.TEXT_PLAIN_VALUE})
+  public ResponseEntity<String> getEffectiveDataRequirements(
+      Principal principal,
+      @RequestBody String bundleStr,
+      @RequestHeader("Authorization") String accessToken,
+      @RequestParam(required = true, name = "libraryName") String libraryName,
+      @RequestParam(required = true, name = "measureId") String measureId) {
```
Comment (adongare) [inline src/main/java/gov/cms/mat/cql_elm_translation/controllers/EffectiveDataRequirementController.java:39]: This is fine for now but maybe we can improve this in the future. we can pass just Measure resource or expression list and CQL string only. we do not need the whole bundle.

**Comment by adongare** [comment]
Can we do following 2 things:
1. remove madie-java-models dependency from pom xml. Don’t think we need it anymore as we are accepting bundle instead of madie measure
2. Earlier we were not aware of effectiveDataRequirements [not having contents](http://hl7.org/fhir/us/cqfmeasures/StructureDefinition-module-definition-library-cqfm.html). please remove `effectiveDataRequirements.setContent(createAttachment(library.getContent()));` https://github.com/MeasureAuthoringTool/cql-elm-translation/blob/2e9caf9109518ff5a89aadfdd9599413f9e4c691/src/main/java/gov/cms/mat/cql_elm_translation/service/EffectiveDataRequirementService.java#L118

File: src/main/java/gov/cms/mat/cql_elm_translation/controllers/EffectiveDataRequirementController.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/mat/cql_elm_translation/controllers/EffectiveDataRequirementController.java:39]: @adongare get effective data requirements seems to need both measure and library?

Comment (adongare) [reply src/main/java/gov/cms/mat/cql_elm_translation/controllers/EffectiveDataRequirementController.java:39]: it needs just expressions(populations, SDEs, stratifications) which is part of measure resource and CQL. So, either we can pass measure resource and CQL or just list of these expressions and measure cql. measure cql will be translated by the translator and fetch included libraries. then build the effectiveDr based on expressions. but let's not handle it now. will create a ticket for it.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/mat/cql_elm_translation/controllers/EffectiveDataRequirementController.java:39]: Ok I see, thanks so much for the info! @adongare

**Review by jkotanchik-SB — APPROVED** [review]
